src/test/java/rx/CompletableTest.java

```diff
@@ -2827,14 +2827,27 @@ public void call(Integer integer) {
         });
     }
 
-    @Test(expected = OnErrorNotImplementedException.class)
```

Comment:
This is a breaking change that can break lots of unit tests and change runtime behavior

Reply:
The issue reported an inconsistency: synchronous crash would throw synchronously but an asynchronous crash would get swallowed.

Reply:
That is expected for 1.x, this is how it always worked

On Android this will crash the app btw, because default Thread.UncaughtExceptionHandler prints exception and exits the app with dialog

User can always hook into that by overriding default Thread.UncaughtExceptionHandler

Reply:
In this case, there can't be a fix for the issue reported in #5237; I suggest **wont-fix** there.

Reply:
We still can try to sync try/catch logic with Observable's `SafeSubscriber.onCompleted`, basically do something like:

```
Exceptions.throwIfFatal(e);
RxJavaHooks.onError(e);
throw new OnCompletedFailedException(e.getMessage(), e); // Use different exception type here
```

Reply:
We'd need a public `OnSuccessFailedException` but v1 is in feature freeze. Also I'm not sure if throwing a different exception than what the lambda threw would solve the original problem.

Reply:
Yeah I see, maybe just add throwIfFatal then?

Reply:
And swallow the non-fatal exceptions? That would still break this particular test.

Reply:
Ah, sorry, of course current behavior includes throwing fatals, no changes required then, I guess.